src/raytracer/treetraversal.m

```diff
@@ -139,22 +134,14 @@
             % plane2, point21, point22, point23 correspond to previous planes
             % and points in Array_of_planes and Array_of_points
             plane2 = ArrayOfPlanes(numberOfPlanes,indexPlanesTemporary - 4:indexPlanesTemporary - 1);
-            normal2(1) = plane2(1:3);
+%             normal2(1) = plane2(1:3);
```

Comment:
deleted, but why in the next line we still have (1)? I corrected this error in this version. May be I did something wrong in merge.
For your reference, it should be             normal2 = plane2(1:3); otherwise it would give you error for third order of reflection. We had corrected this error in my previous PR but somehow this error propagates till now.